### PR TITLE
New panel edit:  General options categorisation experiment

### PR DESCRIPTION
**Pull request by dprokop**
Ref #21111
cc @sarlinska @jessover9000 @torkelo 

This brings some of the categorisation posibilities to the general options pane:

Dark theme:
image

Light theme:
image

### CODE REVIEW

**Comment by dprokop** [comment]
Things to discuss:

- [ ] Overall information architecture of the options
- [ ] Categorised labels (Category > Option) contrasts, colors etc - I couldn't find grays combo that I would be satisfied with :/ @jessover9000 )

File: packages/grafana-data/src/types/OptionsUIRegistryBuilder.ts [hunk 1]
```diff
@@ -8,12 +8,13 @@ import { NumberFieldConfigSettings, SelectFieldConfigSettings, StringFieldConfig
 export interface OptionsEditorItem<TSettings, TEditorProps> extends RegistryItem {
   editor: ComponentType<TEditorProps>;
   settings?: TSettings;
+  category?: string[];
```
Comment (torkelo) [inline packages/grafana-data/src/types/OptionsUIRegistryBuilder.ts]: what's the reason for making it an array?

Comment (torkelo) [reply packages/grafana-data/src/types/OptionsUIRegistryBuilder.ts]: ah, see now, in order to support nesting

**Comment by torkelo** [comment]
@dprokop merge synced this with master, looking promising

**Comment by torkelo** [comment]
done experimenting, naming these sections is veeeeery hard 


image

I like grouping color & thresholds together I see the color option being first in that block (because it will influence the threshold editor UI).

File: public/app/plugins/panel/table2/module.tsx [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
           path: 'width',
-          name: 'Column width',
+          name: 'Width',
           description: 'column width (for table)',
           settings: {
             placeholder: 'auto',
             min: 20,
             max: 300,
           },
+          category: ['Column'],
```
Comment (torkelo) [inline public/app/plugins/panel/table2/module.tsx:18]: not sure this category makes sense, all field options will be column options :)  and there are only two.

**Review by torkelo — APPROVED** [review]
I would remove the sub categories for the table panel 

The naming here is a bit off as all are standard (except the custom):

image

**Comment by jessover9000** [comment]
I do think the added categories might make sense and could help skip some descriptions, but it really depends on the naming of the accordions. I can't really give a qualified opinion on that, would have to see an information architecture of how the properties could be sorted or are sorted and not sure that exists … but I'm sure you'll find a good way to solve it :)

If we do want the categories, color options that I see would be:
1. use #7b8087 (description color) also for category and make the actual field title #c7d0d9 (text-strong color)
2. light up the normal text color for dark theme slightly so that there's more contrast to the description grey
3. keep it the way it is now

I tried making the description and category color darker, but didn't like it because it looks disabled.

**Pull request merged**